scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -119,6 +113,53 @@ object TypedPipe extends Serializable {
       override def hash(x: Int): Int = x
     }
   }
+
+   // TODO: CascadingPipe is still tightly bound to cascading
+   case class CascadingPipe[T](pipe: Pipe,
+    fields: Fields,
+    @transient localFlowDef: FlowDef, // not serializable.
+    @transient mode: Mode,
+    converter: TupleConverter[T]) extends TypedPipe[T]
+
+   case class CrossPipe[T, U](left: TypedPipe[T], right: TypedPipe[U]) extends TypedPipe[(T, U)] {
+     def viaHashJoin: TypedPipe[(T, U)] =
```

Comment:
This transformation seems much simpler than the previous implementations of `cross`. Is it a safe change? Would it be worth keeping the old behavior to reduce the scope of the change?